src/components/task-list-component.jsx

```diff
@@ -18,32 +22,53 @@ export function TaskList() {
         let newItemList = [...itemList];
         const currentTask = newItemList.find((current) => current.id ===id);
         
-        currentTask.isCompleted = true;
+        currentTask.isCompleted = !currentTask.isCompleted;
         setItemList(newItemList);
     }
 
     function deleteCompletedTask() {
-        let newItemList = [...itemList];
-        const notCompletedTasks = getNotCompletedTasks(newItemList)
+        if(itemList.length > 0) {
+            let newItemList = [...itemList];
+            const notCompletedTasks = getNotCompletedTasks(newItemList)
 
-        setItemList(notCompletedTasks);
+            if(haveCompletedTasks(newItemList)){
+                setItemList(notCompletedTasks);
+                setItemListHistory(newItemList);
+                setShowConfirmation(true);
+            }
+        }
+    }
+
+    const confirmDelete = () => {
```

Comment:
un tip que me dio un amigo, las arrow function en estos casos son al pedo, lo mejor en este caso es usar un simple function. Las arrow function se empezar a usar para evitar bindear las funciones en los prototipos por que cuando ponias function no sabia cual era el this.
https://www.freecodecamp.org/news/when-and-why-you-should-use-es6-arrow-functions-and-when-you-shouldnt-3d851d7f0b26/